rundeckapp/grails-app/services/rundeck/services/LogFileStorageService.groovy

```diff
@@ -1184,7 +1184,7 @@ class LogFileStorageService
             boolean checkPartial = false
     )
     {
-        File file = getFileForExecutionFiletype(execution, filetype, false, false)
+        File file = getFileForExecutionFiletype(execution, filetype, execution.scheduledExecution == null, false)
```

Comment:
Use a variable for all these `execution.scheduledExecution == null` in different calls, (use a good name that shows why you need it)

Reply:
We are not supposed to get "Execution log not found after job deletion" since the goal of the card was to show the logs even after a job was deleted. This is possible because Rundeck is saving all executions pointing to where the logs are located.

Reply:
What I mean by covering that use case is to make a test that verifies if the logs are still accessible after deletion trying to use the interfaces that the application is currently using to show them on the screen.

Reply:
Osman try to save this operation `execution.scheduledExecution == null` in a variable because It just like that it didn’t explain why is there.

Reply:
this scenario would still be possible if someone just deletes the log file manually, but i think that is covered by the tests 

https://github.com/rundeck/rundeck/blob/fff68b5a30fd3d06b611e011e8e19ece38b14c1a/rundeckapp/src/test/groovy/rundeck/services/LogFileStorageServiceSpec.groovy#L802

Reply:
Please confirm that with the current tests, in case the logs of deleted jobs don't appear again as before, the tests will complain about it.